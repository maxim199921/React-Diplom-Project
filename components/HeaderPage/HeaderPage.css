.headerLink {
    text-align: center;
    background-color: #130e27;
    background-image: url("../../img/header-background.png");
    width: 100%;
    min-height: 120px;
    position: fixed;
    z-index: 1000;
    top: 0;
    left: 0;
    background-position: center center;
    background-size: cover;
    background-repeat: no-repeat;
}

.headerLink-Logo {
    background-image: url("../../img/LogoMain.png");
    background-repeat: no-repeat;
    min-width: 207px;
    min-height: 72px;
}

.headerNav {
    min-height: 120px;
}

.btn-nav {
    font-size: 18px;
    line-height: 22px;
    color: white;
    margin-left: 16px;
    padding: 5px 14px;
}

.btn-nav:hover {
    background-color: rgba(93, 187, 193, 0.7);
    border-radius: 6%;
}

.btn-nav:hover {
    color: white;
    text-decoration: none;
}

.ActivePageLink {
    background-color: #5dbbc1;
    border-radius: 6%;
}

.ActivePageLink:hover {
    background-color: rgba(93, 187, 193, 1);
}

.btn-basket {
    margin-left: 60px;
}

.headerLink-mainNav {
    text-align: right;
}

@media (max-width: 767px) {
    .headerLink-Logo {
        display: none;
    }
}

@media (max-width: 550px) {
    .headerLink .headerLink-mainNav {
        display: flex;
        flex-direction: column;
        text-align: center;
        margin: 0 auto;
        padding: 0;
        width: 200px;
    }
    .btn-basket {
        margin-left: 0;
    }
    .btn-nav {
        font-size: 14px;
        line-height: 16px;
        color: white;
        margin-left: 0;
        padding: 5px 14px;
    }
    .ActivePageLink {
        background-color: #5dbbc1;
        border-radius: 3%;
        font-weight: bold;
    }
}