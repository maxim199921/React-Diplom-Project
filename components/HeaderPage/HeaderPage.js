import React from 'react';
import {NavLink} from 'react-router-dom';

import './HeaderPage.css';

class HeaderPage extends React.Component {

    render() {

        return (
            <div className="headerLink">
                <div className="container">
                    <div className="row no-gutters align-items-center headerNav">
                        <div className="col-md-4 headerLink-Logo"/>
                        <div className="col-md-8 headerLink-mainNav">
                            <NavLink to="/" exact className='btn-nav' activeClassName="ActivePageLink">Home</NavLink>
                            <NavLink to="/catalogpage" exact className='btn-nav' activeClassName="ActivePageLink">Catalog</NavLink>
                            <NavLink to="/servicespage" className='btn-nav' activeClassName="ActivePageLink">Services</NavLink>
                            <NavLink to="/basketpage" className='btn-nav btn-basket' activeClassName="ActivePageLink">Basket</NavLink>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default HeaderPage;
