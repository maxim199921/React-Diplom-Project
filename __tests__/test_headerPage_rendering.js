import React from 'react';
import renderer from 'react-test-renderer';
import HeaderPage from '../components/HeaderPage/HeaderPage';
import {HashRouter, hashHistory} from 'react-router-dom';

test('тест на рендеринг хедера', () => {

    const component = renderer.create(
        <HashRouter history={hashHistory}>
            <HeaderPage/>
        </HashRouter>
    );

    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

});