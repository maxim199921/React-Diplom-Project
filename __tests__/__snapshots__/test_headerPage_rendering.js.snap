// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`тест на рендеринг хедера 1`] = `
<div
  className="headerLink"
>
  <div
    className="container"
  >
    <div
      className="row no-gutters align-items-center headerNav"
    >
      <div
        className="col-md-4 headerLink-Logo"
      />
      <div
        className="col-md-8 headerLink-mainNav"
      >
        <a
          aria-current="page"
          className="btn-nav ActivePageLink"
          href="#/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="btn-nav"
          href="#/catalogpage"
          onClick={[Function]}
        >
          Catalog
        </a>
        <a
          aria-current={null}
          className="btn-nav"
          href="#/servicespage"
          onClick={[Function]}
        >
          Services
        </a>
        <a
          aria-current={null}
          className="btn-nav btn-basket"
          href="#/basketpage"
          onClick={[Function]}
        >
          Basket
        </a>
      </div>
    </div>
  </div>
</div>
`;
